### DISCUSSION TITLE
Communicating to stock assessment authors about new GOA stratification

### DISCUSSION THREAD

**Discussion by MargaretSiple-NOAA** [afsc-gap-products/StationAllocationAIGOA#19] [Ideas]
Hi team,
Since we've had a few different email convos about this, I thought it might help to consolidate them in a discussion here so we can plan (and selfishly so I can keep track of what we talked about and what our plan is).  I think it would be wise to produce a summary for the assessment authors about the changes and how it might affect our estimates and their workflow. I think Zack has some of this material already, and I would be happy to facilitate the creation of this! I think it's important.

Differences in the products we have for the assessment authors, per Ned: 
> 1 - we'll no longer produce Gulf-wide, depth stratum based biomass or size(length)comp estimates
> 2 - depth strata will generally be unique to a management zone
> 3 - we'll be using the NMFS regulatory area boundaries as management zone strata instead of the historic INPFC statistical areas*
> 4 - we'll be using updated bathymetry to circumscribe the areas of the depth strata

## New tables in Oracle
People will not have seen these before. Everything with `_NMFSAREA` in them will replace `_INPFCAREA` tables in the current GOA schema.
- `BIOMASS_NMFSAREA` (by NMFS management zone)
- `BIOMASS_NMFSAREA_DEPTH` (Depth Strata nested within NMFS management zone)
- `SIZECOMP_NMFSAREA` (by NMFS management zone)
- `SIZECOMP_NMFSAREA_DEPTH` (Depth Strata nested within NMFS management zone)
- `AGECOMP_BY_LENGTH`

## Modified tables in Oracle
- `BIOMASS_TOTAL` (Gulf-wide)
- `SIZECOMP_TOTAL` (Gulf-wide)
- `AGECOMP_TOTAL`

## Table objects no longer supported

- `BIOMASS_DEPTH`
- `BIOMASS_INPFC`
- `BIOMASS_INPFC_DEPTH`
- `BIOMASS_STRATUM`
- `SIZECOMP_INPFC`
- `SIZECOMP_INPFC_DEPTH`
- `SIZECOMP_STRATUM`

## Tables we will need to recreate or modify (may or may not matter to stock assessment authors)

- `GOA_GIS_ATTRIBS`
- `GOAGRID_GIS`
- `GOAGRID`
- `GOA_STRATA`

## Team to do list for communicating rollout 

- [x]  Figure out infrastructure changes to Oracle tables (Ned, with help from Megsie / Zack) - if these changes are insurmountable in the amount of time we have, we may have to delay. Not sure if this is the case at all! Just putting it in as a possibility
- [x]  Make communication materials (slides? Markdown report thing?) for stock assessment team (Megsie with help from Ned/Zack/possibly stock assessment ppl)

# How to communicate these changes 📨
- Side by side comparison of the old and new stratum tables for the Gulf including changes to stratum areas
- Map showing the change in management zone boundaries. Does a table of stratum definitions for the restratified survey already exist somewhere? (Q for @zoyafuso-NOAA )
- Emphasize reduction in bias for area-specific indices (per Stan)

**Comment by Ned-Laman-NOAA** [comment]
Update and catch up on this topic:

After we held a summit with Stock Assessment late in '23 and had some additional conversations among ourselves, we decided to delay implementing the Gulf restratification until 2025. This gives stock assessment ample time to revise their data access code to point at and handle the new table structures and gives us more time to work out the details of changing infrastructure, etc.

We committed to producing a mock restratified data set in new data structures for stock authors to test in Spring of 2023-nominally making the table objects available by the end of Feb. '23.

Recent developments (2/3/23):

On a parallel trajectory, the Data Processes Working Group (DPWG) and Index Computation Working Group (ICWG) are moving toward a future with unified GAP bottom trawl survey data product tables in the GAP_PRODUCTS schema. It is our Program's intention that these unified products will replace the dispersed datascape we now serve to stock assessment (e.g., products in different schemata like GOA, HAEHNR, and EBSSHELF).

With the above in mind, I propose that we produce the mock restratified data promised for stock author testing in the form and shape of the new, unified data product tables being developed for GAP_PRODUCTS.

Challenge: Those tables are not fully developed yet and will need vetting by an Oracle DBA before they're ready for production. This may cause our timeline to slip.

Interested in any thoughts you all have...

**Reply by MargaretSiple-NOAA** [reply]
I think it is a good idea to put the new tables in that proposed new schema, though I'm a little but useless on this point because I don't really understand Oracle table infrastructure. Regarding the tables needing vetting by an Oracle DBA, is that Heather? Has Heather (or similar extremely database-savvy person) generally been a part of the data processes conversation? 

 I keep thinking about those new tables and how they will affect what everyone else is doing (stock assessment data pull/process group, restratification effort, etc). I think I'm not 100% convinced yet that restructuring the existing tables (haul, catch, the basics) will improve peoples' workflows, but I **do** like the idea of a brand new schema for pushing some of these new tables. Just my $0.02 at this particular moment.

**Comment by Ned-Laman-NOAA** [comment]
Expanding on Megsie's response:

Having Heather give us some tips once we're a little closer to some final table structures for the consolidated products table makes a ton of sense; wonder if she has time. DBA review would come from OFIS and the officially trained DBAs there. I'm probably the most database savvy person in the DPWG conversation but haven't seen an actual table structure to comment on just yet. Might happen tomorrow when I meet with Emily.

Trying to make the hazy picture a little clearer as far as what we're trying to do. 
RACEBase and its tables (e.g., CATCH, HAUL, LENGTH, SPECIMEN) will remain unchanged.
Products based on those tables (e.g., CPUE, AGECOMP, or BIOMASS) are what we are looking to consolidate into GAP_PRODUCTS. 
There are a couple of avenues of discussion right now for how this will go. One camp holds that we can have a single BIOMASS table that would contain all biomass flavors (e.g., Bering Sea Biomass by Stratum or Gulf of Alaska Biomass by Management Area) instead of our present paradigm where there's a BIOMASS_STRATUM table and a BIOMASS_AREA table. I'm not convinced that one table to rule them all is best database practices or very efficient, but am withholding judgement until I see what Emily envisions there. 

Hopefully that last bit was helpful rather than confusing.
